tests/functional/UseCase/Bucket/ShareLinkBucketTest.php

```diff
@@ -0,0 +1,359 @@
+use Keboola\StorageDriver\BigQuery\Handler\Bucket\Share\ShareBucketHandler;
+use Keboola\StorageDriver\BigQuery\Handler\Bucket\UnLink\UnLinkBucketHandler;
+use Keboola\StorageDriver\BigQuery\Handler\Bucket\UnShare\UnShareBucketHandler;
+use Keboola\StorageDriver\BigQuery\Handler\Table\Create\CreateTableHandler;
+use Keboola\StorageDriver\Command\Bucket\LinkBucketCommand;
+use Keboola\StorageDriver\Command\Bucket\LinkedBucketResponse;
+use Keboola\StorageDriver\Command\Bucket\ShareBucketCommand;
+use Keboola\StorageDriver\Command\Bucket\ShareBucketResponse;
+use Keboola\StorageDriver\Command\Bucket\UnlinkBucketCommand;
+use Keboola\StorageDriver\Command\Bucket\UnshareBucketCommand;
+use Keboola\StorageDriver\Command\Project\CreateProjectResponse;
+use Keboola\StorageDriver\Command\Table\CreateTableCommand;
+use Keboola\StorageDriver\Credentials\GenericBackendCredentials;
+use Keboola\StorageDriver\FunctionalTests\BaseCase;
+use Keboola\TableBackendUtils\Escaping\Bigquery\BigqueryQuote;
+
+class ShareLinkBucketTest extends BaseCase
+{
+    protected GenericBackendCredentials $sourceProjectCredentials;
+    protected CreateProjectResponse $sourceProjectResponse;
+
+    protected GenericBackendCredentials $targetProjectCredentials;
+    protected CreateProjectResponse $targetProjectResponse;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->cleanTestProject();
+
+        [$credentials1, $response1] = $this->createTestProject();
+        $this->projectSuffix = '-s';
+        [$credentials2, $response2] = $this->createTestProject();
+
+        // project1 shares bucket
+        $this->sourceProjectCredentials = $credentials1;
+        $this->sourceProjectResponse = $response1;
+
+        // project2 checks the access
+        $this->targetProjectCredentials = $credentials2;
+        $this->targetProjectResponse = $response2;
+    }
+
+    protected function tearDown(): void
+    {
+        parent::tearDown();
+        $this->cleanTestProject();
+    }
+
+    public function testShareAndLinkBucket(): void
+    {
+        $bucketResponse = $this->createTestBucket($this->sourceProjectCredentials);
+
+        $bucketDatabaseName = $bucketResponse->getCreateBucketObjectName();
+
+        $sourceBqClient = $this->clientManager->getBigQueryClient($this->sourceProjectCredentials);
+        $linkedBucketSchemaName = $bucketDatabaseName . '_LINKED';
+
+        $handler = new CreateTableHandler($this->clientManager);
+        $path = new RepeatedField(GPBType::STRING);
+        $path[] = $bucketDatabaseName;
+        $columns = new RepeatedField(GPBType::MESSAGE, CreateTableCommand\TableColumn::class);
+        $columns[] = (new CreateTableCommand\TableColumn())
+            ->setName('ID')
+            ->setType(Bigquery::TYPE_INTEGER);
+        $command = (new CreateTableCommand())
+            ->setPath($path)
+            ->setTableName('TESTTABLE_BEFORE')
+            ->setColumns($columns);
```

Comment:
konstanta na ten nazev tabulky?

Reply:
v testoch je to asi jedno resp mozno je to lepsie ze to musis rucne upravt na kazdom mieste zvlast. Kazdopadne ked uz som an to hrabal tak som to dal do const